presto-main/src/main/java/com/facebook/presto/operator/RemoteProjectOperator.java

```diff
@@ -104,7 +107,8 @@ public Page getOutput()
             Page output;
             try {
                 for (int i = 0; i < blocks.length; i++) {
-                    blocks[i] = result[i].get();
+                    blocks[i] = result[i].get().getResult();
+                    operatorContext.recordAdditionalCpu(MILLISECONDS.toNanos(result[i].get().getCpuTimeMs()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using CompletableFuture.get() without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-).